pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/UuidPreProcessor.java

```diff
@@ -0,0 +1,69 @@
+package com.redhat.pantheon.asciidoctor.extension;
+
+import org.asciidoctor.ast.Document;
+import org.asciidoctor.extension.Preprocessor;
+import org.asciidoctor.extension.PreprocessorReader;
+import com.redhat.pantheon.jcr.JcrQueryHelper;
+import org.apache.sling.api.resource.ResourceResolver;
+import java.util.Optional;
+
+import org.apache.sling.api.resource.Resource;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UuidPreProcessor extends Preprocessor {   // (1)
+
+    private static Resource module;
+    private static final Logger log = LoggerFactory.getLogger(UuidPreProcessor.class);
+    private static String newModulePath;
+
+    public UuidPreProcessor(Resource module) {
+        this.module = module;
+    }
+
+    @Override
+    public void process(Document document, PreprocessorReader reader) {
+        
+        List<String> lines = reader.readLines(); 
+        List<String> newLines = new ArrayList<String>();
+        String[] split;
+        String uuid, newLink;
+        
+        for (String line: lines) {
+            if(line.startsWith("xref:")){
+                split = line.split(",pantheon-id=");
+                uuid = split[1].replace(split[1].substring(split[1].length()-1), "");                
+                resolveActualPath(module.getResourceResolver(), uuid);
+                newLink = split[0].replaceAll(":.*?\\[", ":"+newModulePath+"[");                
+                newLink = newLink + "]";
+                newLines.add(newLink);
+            }else {
+                    newLines.add(line);
+            }
+        }
+        reader.restoreLines(newLines);
+    }
+
+    private static void resolveActualPath(ResourceResolver resolver, String uuid) {
+        JcrQueryHelper qh = new JcrQueryHelper(resolver);        
+        try {
+                Optional<Resource> result = 
+                    qh.query("select * from [nt:base] WHERE [jcr:uuid] = '" + uuid + "'")
```

Comment:
I know we talked about having `[nt:base]` here earlier. It may not make a difference from a performance perspective, but from a functionality perspective, it makes sense to put `[pant:module]` here instead, defensively. The reason is that we may expose UUIDs for other objects (maybe products, for some reason?) to authors at some point in the future. If the author accidentally supplies a product's UUID in an xref, we wouldn't want this code resolving that. It is better to specify that we are looking for modules only and to return no result in such a scenario.

Reply:
This makes sense, since this is searching for all the UUIDs across the hierarchy, we might have to switch to [pant:module]